src/components/UI/SearchInput.js

```diff
@@ -0,0 +1,53 @@
+import styles from "./SearchInput.module.css";
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+
+const DEFAULT_DELAY = 1000;
+
+const SearchInput = ({ onChange, delay }) => {
+  const [value, setValue] = useState("");
+  const [isEmpty, setIsEmpty] = useState(true);
+  const timeoutId = useRef();
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeoutId.current);
+    };
+  }, []);
+
+  const changeWithDebounce = (value) => {
```

Comment:
This function will be created as new instance on every render...

Think about useCallback() or refactor code